In [1]:
%run data_gather-Copy1.ipynb

{'_type': 'album',
 'api_path': '/albums/102298',
 'cover_art_thumbnail_url': 'https://images.genius.com/dc9148a9c2eb963ae0b5564ef2042412.300x300x1.jpg',
 'cover_art_url': 'https://images.genius.com/dc9148a9c2eb963ae0b5564ef2042412.1000x1000x1.jpg',
 'full_title': 'Euphoria Mourning by Chris Cornell',
 'id': 102298,
 'name': 'Euphoria Mourning',
 'name_with_artist': 'Euphoria Mourning (artist: Chris Cornell)',
 'primary_artist_names': 'Chris Cornell',
 'pusher_channel': 'album-102298',
 'release_date_components': {'year': 1999, 'month': 9, 'day': 21},
 'release_date_for_display': 'September 21, 1999',
 'updated_at': 1739842718,
 'url': 'https://genius.com/albums/Chris-cornell/Euphoria-mourning',
 'current_user_metadata': {'permissions': ['create_comment'],
                           'excluded_permissions': ['set_album_color_gradient',
                                                    'edit',
                                                    'destroy',
                              

In [2]:
%run data_cleaning-no-labels.ipynb

{'_type': 'album',
 'api_path': '/albums/102298',
 'cover_art_thumbnail_url': 'https://images.genius.com/dc9148a9c2eb963ae0b5564ef2042412.300x300x1.jpg',
 'cover_art_url': 'https://images.genius.com/dc9148a9c2eb963ae0b5564ef2042412.1000x1000x1.jpg',
 'full_title': 'Euphoria Mourning by Chris Cornell',
 'id': 102298,
 'name': 'Euphoria Mourning',
 'name_with_artist': 'Euphoria Mourning (artist: Chris Cornell)',
 'primary_artist_names': 'Chris Cornell',
 'pusher_channel': 'album-102298',
 'release_date_components': {'year': 1999, 'month': 9, 'day': 21},
 'release_date_for_display': 'September 21, 1999',
 'updated_at': 1739842718,
 'url': 'https://genius.com/albums/Chris-cornell/Euphoria-mourning',
 'current_user_metadata': {'permissions': ['create_comment'],
                           'excluded_permissions': ['set_album_color_gradient',
                                                    'edit',
                                                    'destroy',
                              

In [3]:
chris_cornell_df_without_labels.reset_index(drop=True)

,track_title,track_lyrics,album_name,project_name,release_year,label,processed_track_lyrics
0,Can’t Change Me,"[Verse 1] She can do anything at all Have anything she pleases The power to change what she thinks is wrong But what could she want with me? Yeah [Pre-Chorus] But wait just one minute here I can see that she's trying to read me (Suddenly I know) [Chorus] She's going to change the world She's going to change the world She's going to change the world But she can't change me No, she can't change me [Verse 2] She has the daylight at her command She gives the night its dreams, yeah She can uncover your darkest fear And make you forget you feel it [Pre-Chorus] But wait just one minute more I can see that she's trying to free me (Suddenly I know) [Chorus] She's going to change the world (Going to change the world) She's going to change the world (Going to change the world) She's going to change the world (Going to change) But she can't change me No, she can't change me [Bridge] And suddenly I can see everything that's wrong with me, yeah But what can I do? I'm the only thing I really have at all [Instrumental Break] [Pre-Chorus] But wait, just, one minute here I can see that she's trying to need me (Suddenly I know) [Chorus] She's going to change the world (Going to change the world) She's going to change the world (Going to change the world) She's going to change the world (Going to change) But she can't change me No, she can't change me",Euphoria Mourning,Chris Cornell,1999,A&M Records,she do she pleases power change she thinks wrong she want me ? wait minute i that she's trying read me suddenly i know she's going change world she's going change world she's going change world she can't change me no she can't change me she daylight her command she gives night dreams she uncover darkest fear you forget you feel it wait minute i that she's trying free me suddenly i know she's going change world going change world she's going change world going change world she's going change world going change she can't change me no she can't change me suddenly i that's wrong me i do ? i'm thing i break wait minute i that she's trying need me suddenly i know she's going change world going change world she's going change world going change world she's going change world going change she can't change me no she can't change me
1,Flutter Girl,"[Verse 1] I'm drinking dust with eyes of rust Tonight my tears might stain your wings So flutter home because you're better off alone than with me [Verse 2] So hide your face and tie your lace And butterflies across your cheek Forget how soon you become a fool for words when I speak [Chorus] Flutter girl You don't want to know what I live You don't want to take what I give Because I give nothing for free You don't want to know what I live You don't want to take what I give Because I give nothing for free [Verse 3] My hands are rough, my fingers cold And your heart's so young and so naive To ever feel for a moment that I might dare to believe [Chorus] Flutter girl You don't want to know what I live You don't want to take what I give Because I give nothing for free You don't want to know what I live You don't want to take what I give Because I give nothing for free [Bridge] I'll tear your wings As I melt in your smile As I run all your colors away [Chorus] Flutter girl You don't want to know what I live You don't want to take what I give Because I give nothing for free You don't want to know what I live You don't want to take what I give Because I give nothing for free [Outro] Flutter girl",Euphoria Mourning,Chris Cornell,1999,A&M Records,i'm drinking dust eyes rust tonight tears stain wings flutter home you're better me hide face tie lace butterflies cheek forget soon you fool words i speak flutter girl you don't want know i live you don't want i i free you don't want know i live you don't want i i free hands rough fingers cold heart's young naive feel moment that i dare believe flutter girl you don't want k

In [4]:
import re
from collections import Counter

text = chris_cornell_df_without_labels.loc[chris_cornell_df_without_labels["album_name"] == "Euphoria Mourning", "processed_track_lyrics"]

words = text.str.split(" ").sum()

Counter(words).most_common(30)

[('you', 110),
 ('i', 95),
 ("i'm", 55),
 ("it's", 30),
 ('me', 25),
 ('know', 25),
 ('world', 24),
 ('change', 22),
 ('that', 19),
 ('want', 17),
 ("you're", 17),
 ('rain', 17),
 ('going', 16),
 ("don't", 16),
 ('sweet', 15),
 ('steel', 15),
 ('she', 13),
 ('like', 13),
 ('taking', 13),
 ("she's", 12),
 ('way', 12),
 ('come', 12),
 ("can't", 11),
 ('it', 11),
 ('away', 11),
 ('love', 11),
 ('no', 10),
 ('today', 10),
 ('alright', 10),
 ('falling', 10)]

QUESTION 1a: How did Chris Cornell’s use of pronouns change when looking at the first solo album versus last solo album?

First solo album: Euphoria Mourning
[('you', 110),
 ('i', 95),
 ("i'm", 55),
 ("it's", 30),
 ('me', 25),
 ("you're", 17),
 ('she', 13),
 ("she's", 12),
 ('it', 11),
 ("we're", 7),
 ('we', 5),
 ('us', 3)

Question 2a: What are the top 10 nouns Chris Cornell uses in each album. How many of them are concrete and how many are abstract? 

First solo album: Euphoria Morning 

NOTE: combined i, i'm, and me even though one's in the subjective and one's in the objective since they're referring to the same noun 


('you', 110), ("you're", 17),
 ('i', 95), ("i'm", 55), ('me', 21),
 ('world', 24),
 ('change', 22),
 ('steel', 15),
 ('she', 13),("she's", 12),
 ('way', 12),
 ('rain', 12),
 ('love', 11),
('mission', 10),]

In [5]:
# text = chris_cornell_df_without_labels.loc[chris_cornell_df_without_labels["album_name"] == "Higher Truth", "processed_track_lyrics"]

# words = text.str.split(" ").sum()

# counts=Counter(words)

# counts_df = pd.DataFrame(
#     counts.items(),
#     columns=["word", "count"]
# )

# counts_df = counts_df.sort_values("count", ascending = False)

In [6]:
first = ["i", "i'm", "me", "my", "mine", "myself",
         "we", "us", "our", "ours", "ourselves"]

second = ["you", "your", "yours", "yourself", "yourselves"]

third = ["he", "he's", "him", "his", "himself",
         "she", "she's", "her", "hers", "herself",
         "it", "it's", "its", "itself",
         "they", "them", "their", "theirs", "themselves"]

In [16]:
#Create new dataframe for pronoun counts per album and by each pronoun person (first, second, third)

album_data = chris_cornell_df_without_labels[
    chris_cornell_df_without_labels["album_name"].isin(
        ["Euphoria Mourning",
         "Higher Truth",
         "Ultramega OK",
         "King Animal",
         "Audioslave",
         "Revelations",
        "Temple of the Dog"])]

counts = []

for album, group in album_data.groupby("album_name"):
    album_words = " ".join(group["processed_track_lyrics"]).split()
    counter = Counter(album_words)
    
    counts.append({
        "album": album,
        "pronoun": "First",
        "pronoun_count": sum(counter[first_person] for first_person in first)
    })
    
    counts.append({
        "album": album,
        "pronoun": "Second",
        "pronoun_count": sum(counter[second_person] for second_person in second)
    })
    
    counts.append({
        "album": album,
        "pronoun": "Third",
        "pronoun_count": sum(counter[third_person] for third_person in third)
    })


pronoun_count_df = pd.DataFrame(counts)

pronoun_count_df

,album,pronoun,pronoun_count
0,Audioslave,First,224
1,Audioslave,Second,169
2,Audioslave,Third,86
3,Euphoria Mourning,First,183
4,Euphoria Mourning,Second,113
5,Euphoria Mourning,Third,69
6,Higher Truth,First,270
7,Higher Truth,Second,122
8,Higher Truth,Third,115
9,King Animal,First,195


In [17]:
euphoria_higher = pronoun_count_df[
    pronoun_count_df["album"].isin(["Euphoria Mourning", "Higher Truth"])
]

ultramega_king = pronoun_count_df[
    pronoun_count_df["album"].isin(["Ultramega OK", "King Animal"])
]

audioslave_revelations = pronoun_count_df[
    pronoun_count_df["album"].isin(["Audioslave", "Revelations"])
]

In [18]:
#Create new lists to add as new columns to new dataframe 

word_count_processed_lyrics = []

text = chris_cornell_df_without_labels['processed_track_lyrics']

for text in chris_cornell_df_without_labels["processed_track_lyrics"]:
    tokens = text.split()
    word_count_processed_lyrics.append(len(tokens))

chris_cornell_df_without_labels["word_count_processed_lyrics"] = word_count_processed_lyrics

In [19]:
total_word_count_euphoriamourning = chris_cornell_df_without_labels.loc[chris_cornell_df_without_labels['album_name'] == 'Euphoria Mourning', 'word_count_processed_lyrics'].sum()

total_word_count_highertruth = chris_cornell_df_without_labels.loc[chris_cornell_df_without_labels['album_name'] == 'Higher Truth', 'word_count_processed_lyrics'].sum()

total_word_count_ultramegaok = chris_cornell_df_without_labels.loc[chris_cornell_df_without_labels['album_name'] == 'Ultramega OK', 'word_count_processed_lyrics'].sum()

total_word_count_kinganimal = chris_cornell_df_without_labels.loc[chris_cornell_df_without_labels['album_name'] == 'King Animal', 'word_count_processed_lyrics'].sum()

total_word_count_audioslave = chris_cornell_df_without_labels.loc[chris_cornell_df_without_labels['album_name'] == 'Audioslave', 'word_count_processed_lyrics'].sum()

total_word_count_revelations = chris_cornell_df_without_labels.loc[chris_cornell_df_without_labels['album_name'] == 'Revelations', 'word_count_processed_lyrics'].sum()

total_word_count_temple = chris_cornell_df_without_labels.loc[chris_cornell_df_without_labels['album_name'] == 'Temple of the Dog', 'word_count_processed_lyrics'].sum()


1535 euphoria morrning
2280 higher truth
908 ultramegaok
1483 kinganimal
1569 audioslave
1545 revelations 
1375 temple

In [20]:
word_count_totals = {
    'Euphoria Mourning': 1535,
    'Higher Truth': 2280,
    'Ultramega OK': 908,
    'King Animal': 1483,
    'Audioslave': 1569,
    'Revelations': 1545,
    'Temple of the Dog': 1375}

pronoun_count_df['total_word_count'] =  pronoun_count_df['album'].map(word_count_totals)

In [21]:
pronoun_count_df

,album,pronoun,pronoun_count,total_word_count
0,Audioslave,First,224,1569
1,Audioslave,Second,169,1569
2,Audioslave,Third,86,1569
3,Euphoria Mourning,First,183,1535
4,Euphoria Mourning,Second,113,1535
5,Euphoria Mourning,Third,69,1535
6,Higher Truth,First,270,2280
7,Higher Truth,Second,122,2280
8,Higher Truth,Third,115,2280
9,King Animal,First,195,1483


In [22]:
pronoun_count_df['pronoun_percentage'] = (pronoun_count_df['pronoun_count'] / pronoun_count_df['total_word_count'])*100

In [23]:
pronoun_count_df=pronoun_count_df.round(2)

In [24]:
pronoun_bar_chart_df = pronoun_count_df.pivot(
    index="album",
    columns="pronoun",
    values="pronoun_percentage").reset_index()

In [25]:
pronoun_bar_chart_df

pronoun,album,First,Second,Third
0,Audioslave,14.28,10.77,5.48
1,Euphoria Mourning,11.92,7.36,4.50
2,Higher Truth,11.84,5.35,5.04
3,King Animal,13.15,4.65,2.56
4,Revelations,11.26,3.43,3.30
5,Temple of the Dog,12.95,5.75,5.82
6,Ultramega OK,5.95,2.42,8.70


In [26]:
pronoun_bar_chart_df.to_csv("pronoun_count_df.csv")